apps/src/dance/DanceVisualizationColumn.jsx

```diff
@@ -67,27 +74,38 @@ class DanceVisualizationColumn extends React.Component {
   };
 
   state = {
-    filterOff: this.setFilterStatus()
+    filterOn: this.getFilterStatus()
   };
 
   /*
     Turn the song filter off
   */
   turnFilterOff = () => {
-    this.setState({filterOff: true});
+    this.setState({filterOn: false});
   };
 
   /*
     The filter defaults to on. If the user is over 13 (identified via account or anon dialog), filter turns off.
    */
-  setFilterStatus() {
+  getFilterStatus() {
+    const {userType, under13} = this.props;
+
+    // Check if song filter override is triggered and initialize song filter to true.
+    const songFilter = songFilterOn();
+    if (songFilter) {
+      return true;
+    }
+
     // userType - 'teacher', 'student', 'unknown' - signed out users.
-    // under13 - boolean for signed in user representing age category. Teacher assumed > 13.
-    const signedInOver13 =
-      this.props.userType === 'teacher' ||
-      (this.props.userType === 'student' && !this.props.under13);
-    const signedOutAge = signedOutOver13();
-    return signedInOver13 || signedOutAge;
+    // If user is signed out, query session key to determine age.
+    // Return false (no filter), if user is over 13.
+    if (userType === 'unknown') {
+      return !signedOutOver13();
+    }
```

Comment:
Slightly confused by the comment & code layout here.  Presumably `userType === 'unknown'` checks whether a user is signed out, and `signedOutOver13()` queries the session key?  It could be worth shuffling a little to make these things more clear, e.g.

```
    // If user is signed out...
    if (userType === 'unknown') {
      // ...call signedOutOver13() to query session key to determine age.
      // Return false (no filter), if user is over 13.
      return !signedOutOver13();
    }
```

I think there's a second issue here: `signedOutOver13()` should probably be able to return `true` if the user is signed out and over 13, but it sounds like it isn't able to determine whether the user is signed out, so we have to wrap with this extra check first?  It seems weird to split that logic across two places, and means the function isn't able to do everything you'd expect it could.

Reply:
Interesting point on the second issue. After reflecting on it, I realized that the function name currently infers a lot, so I updated the name to more accurately represent what it does - not if the user has selected `over13` in the age dialog - `ageDialogSelectedOver13`.